File: TaskOutArrivalEvent.cpp
//
// Created by Can Candan on 04/06/17.
//

#include "TaskOutArrivalEvent.h"
#include "OutInEvent.h"
#include "easylogging++.h"

TaskOutArrivalEvent::TaskOutArrivalEvent(Simulation *pSimulation, Task *pTask, double time) : Event(pSimulation, pTask, time) {
    name="TaskOutArrivalEvent";
}

void TaskOutArrivalEvent::process() {
    log();
    OutputDevice* out= this->sim->getFirstIdleOut();
    if (out) {
        OutInEvent* event= new OutInEvent(sim, task, time, out);
        sim->schedule(event);
    } else {
        sim->rrQueue.push(this->task);
    }
}

File: TaskOutArrivalEvent.java
/**
 * Created by veraxmedax on 03/06/2017.
 */
public class TaskOutArrivalEvent extends Event {

    public TaskOutArrivalEvent(Simulation sim, Task task, double time) {
        super(sim, task, time);
    }

    @Override
    void process() {
        OutputDevice outputDevice = getSim().getFirstIdleOutputDevice();
        if (outputDevice != null) {
            Event outInEvent = new OutInEvent(getSim(), getTask(), getTime(), outputDevice);
            getSim().schedule(outInEvent);
        } else {
            getSim().getRrQueue().add(getTask());
        }
    }
}
